backend/github.js

```diff
@@ -96,3 +96,75 @@ exports.deleteHook = function (name, hook, accessToken, callback) {
     }
   });
 };
+
+/*
+ * hookValidator is used to check whether hook is registered to repository or not
+ * @param repositoryName: name of the repository
+ * @param token: github access token
+ */
+exports.hookValidator = function (repositoryName, token) {
+  return new Promise(function(resolve, reject) {
+    request({
+      url: `https://api.github.com/repos/${repositoryName}/hooks`,
+      headers: {
+        'User-Agent': 'Yaydoc',
+        'Authorization': 'token ' + crypter.decrypt(token)
+      }
+    }, function (error, response, body) {
+      var isRegistered = false;
+      if (response.statusCode !== 200) {
+        console.log(response.statusCode + ': ' + response.statusMessage);
+        reject()
+      }
+      var hooks = JSON.parse(body);
+      var hookurl = 'http://' + process.env.HOSTNAME + '/ci/webhook';
+      for (var i = 0; i < hooks.length; i++) {
+        if (hooks[i].config.url === hookurl) {
+          isRegistered = true;
+          break;
+        }
+      }
+      resolve({repositoryName: repositoryName, isRegistered: isRegistered})
+    });
+  });
+}
+
+/**
+ * registerHook is used to register hook to the respository
+ * @param repositoryName: name of the repository
+ * @param token: github access token
+ */
+exports.registerHook = function (data, token) {
+return new Promise(function(resolve, reject) {
+  var hookurl = 'http://' + process.env.HOSTNAME + '/ci/webhook';
+  if (data.sub === true) {
+    hookurl += `?sub=true`;
+  }
+  request({
+    url: `https://api.github.com/repos/${data.repositoryName}/hooks`,
+    headers: {
+      'User-Agent': 'Yaydoc',
+      'Authorization': 'token ' + crypter.decrypt(token)
+    },
+    method: 'POST',
+    json: {
+      name: "web",
+      active: true,
+      events: [
+        "push"
+      ],
+      config: {
+        url: hookurl,
+        content_type: "json"
+      }
+    }
+  }, function(error, response, body) {
+    if (response.statusCode !== 201) {
+      console.log(response.statusCode + ': ' + response.statusMessage);
+      resolve({status: false, body:body});
```

Comment:
`Promise.all` will go to catch if we reject. we have to remove the previously registered webhook if any of the webhook fails. I didn't write the function but for the future, I wrote